ui/components-react/src/components-react/newEditors/Types.ts

```diff
@@ -0,0 +1,56 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+import type { ValueMetadata } from "./values/Metadata.js";
+import type { Value } from "./values/Values.js";
+
+/**
+ * An editor specification defining single editor with a predicate that determines if the editor can be used for a given value.
+ * @beta
+ */
+export interface EditorSpec {
+  applies: (metaData: ValueMetadata, value: Value | undefined) => boolean;
+  Editor: React.ComponentType<EditorProps>;
+}
+
+/**
+ * Generic editor props that are supplied to the editor for rendering.
+ * @beta
+ */
+export interface EditorProps<TMetadata = ValueMetadata, TValue = Value> {
+  metadata: TMetadata;
+  value?: TValue;
+  onChange: (value: TValue) => void;
+  onFinish: () => void;
+  disabled?: boolean;
+  size?: "small" | "large";
+}
+
+/**
+ * Utility function to create an editor spec for editor with concrete metadata and value types.
+ * @beta
+ */
+export function createEditorSpec<
+  TMetadata extends ValueMetadata,
+  TValue extends Value
+>({
+  Editor,
+  isMetadataSupported,
+  isValueSupported,
+  applies,
+}: {
+  isMetadataSupported: (metadata: ValueMetadata) => metadata is TMetadata;
+  isValueSupported: (value: Value) => value is TValue;
+  applies?: (metadata: TMetadata, value?: TValue) => boolean;
+  Editor: React.ComponentType<EditorProps<TMetadata, TValue>>;
+}): EditorSpec {
+  return {
+    applies: (metadata: ValueMetadata, value?: Value) =>
+      isMetadataSupported(metadata) &&
+      (value === undefined || isValueSupported(value)) &&
```

Comment:
`applies` will return `true` when `metadata.isNullable === false` and `value === undefined`. Is that intended?

Reply:
Yes, this is intended for use cases where editor is rendered without initial value.

Reply:
It means the editor is responsible for making it clear to users that it requires a value. Do we have a plan for how editors will expose validation errors?

Reply:
Created a new issue for investigating and implementing the ability to show validation errors. https://github.com/iTwin/appui/issues/1219